taskvine/src/worker/vine_sandbox.c

```diff
@@ -54,6 +54,10 @@ vine_cache_status_t vine_sandbox_ensure(struct vine_process *p, struct vine_cach
 			break;
 		case VINE_CACHE_STATUS_UNKNOWN:
 		case VINE_CACHE_STATUS_FAILED:
+			if (p->task->group_id) {
```

Comment:
Re our discussion today, move this failure-handling logic to `task_can_run_eventually`.
If a task has an input file that is `UNKNOWN`, check it see if another known tasks produces it as an output file.
If so, then the task can stay until it becomes runnable.